app/models/settings/smtp.rb

```diff
@@ -0,0 +1,40 @@
+module Settings
+  class SMTP < RailsSettings::Base
+    self.table_name = :settings_smtp
+
+    OPTIONS = %i[address port authentication user_name password domain].freeze
+
+    # The configuration is cached, change this if you want to force update
+    # the cache, or call Settings::Smtp.clear_cache
+    cache_prefix { "v1" }
+
+    field :address, type: :string, default: ApplicationConfig["SMTP_ADDRESS"]
+    field :authentication, type: :string, default: ApplicationConfig["SMTP_AUTHENTICATION"],
+                           validates: { inclusion: %w[plain login cram_md5] }
+    field :domain, type: :string, default: ApplicationConfig["SMTP_DOMAIN"]
+    field :password, type: :string, default: ApplicationConfig["SMTP_PASSWORD"]
+    field :port, type: :integer, default: ApplicationConfig["SMTP_PORT"].presence || 25
+    field :user_name, type: :string, default: ApplicationConfig["SMTP_USER_NAME"]
+
+    class << self
+      def settings
+        return sendgrid_settings if ENV["SENDGRID_API_KEY"].present?
+
+        keys.index_with { |k| public_send(k) }.symbolize_keys
```

Comment:
The amount of methods in `ActiveSupport` never ceases to amaze me. `index_with` was a new one for me. I'd probably have gone for the plain Ruby version:

```ruby
keys.map { |k| [k.to_sym, public_send(k)] }.to_h
```

Reply:
That is actually new to me too! it was the linter that suggested that change to me.